spec/bowlingSpec.js

```diff
@@ -0,0 +1,171 @@
+describe('Bowling', function ()  {
+
+  beforeEach(function() {
+    bowling = new Bowling();
+  });
+
+  describe('#throw', function() {
+    it('Throw the ball to knock over 10 pins', function () {
+      expect(bowling.throw(10)).toEqual(10);
+    });
+  });
+
+  describe('#record', function() {
+    it('Record the score of the throw', function() {
+      bowling.lastScore = 10;
+      bowling.record();
+      expect(bowling.points[1]).toEqual(10)
+    });
+  });
+  describe('#reducePins', function () {
+    it('Reduce pins of the actual frame', function() {
+      bowling.throw(5);
+      bowling.reducePins(5);
+      expect(bowling.frames[bowling.actualFrame].pins).toEqual(5)
+    });
+  });
+  describe('#increaseActualFrame', function () {
+    it('increase actual frame by one if framePoins.length === 2', function () {
+      bowling.recordInFrame(2)
+      bowling.recordInFrame(2)
+      bowling.increaseActualFrame();
+      expect(bowling.actualFrame).toEqual(2)
+    });
+  });
+
+  describe('#recordInFrame', function () {
+    it("The score of the first roll is saved inside the framePoins array", function() {
+      bowling.throw(5);
+      bowling.recordInFrame(5);
+      expect(bowling.frames[bowling.actualFrame].framePoints[0]).toEqual(5)
+    });
+  });
+  describe('#recordInFrame', function () {
+    it("The score of the second roll is saved inside the framePoins array", function() {
+      bowling.throw(5);
+      bowling.recordInFrame(5);
+      bowling.throw(2);
+      bowling.recordInFrame(2);
+      expect(bowling.frames[bowling.actualFrame].framePoints[1]).toEqual(2)
+    });
+  });
+  describe('#reducePins', function () {
+    it("reduce the pins of a given frame.", function() {
+      bowling.reducePins(5);
+      expect(bowling.frames[bowling.actualFrame].pins).toEqual(5)
+    });
+  });
+  describe('#spareBonus', function () {
+    it("if in the previous frame you did spare, the first score of the next frame is doubled ", function() {
+      bowling.throw(5);
+      bowling.recordInFrame(5);
+      bowling.reducePins(5)
+      bowling.increaseActualFrame();
+      bowling.throw(5);
+      bowling.recordInFrame(5);
+      bowling.reducePins(5)
+      bowling.increaseActualFrame();
+      bowling.throw(5);
+      bowling.recordInFrame(5);
+      bowling.spareBonus();
```

Comment:
If I have to call a lot of methods on my test object in order to achieve the right test setup, I see that as a sign that my code either has a lot of tight coupling or that I haven't effectively modelled the behaviour.

For example, in this case:
- 'throw', 'recordInFrame' and 'reducePins' are all the result of the same action. I would expect to be able to call 'throw' and for 'recordinframe' and 'reducePins' to happen automatically. The client calling bowling shouldn't need to know how to do those things - bowling needs to know that when we throw down some pins, we also need to record the score and reduce the number of pins at the target.
- 'increaseActualFrame' is a a game rule rather than a direct result of the throw, but it's also something that we shouldn't need to know about. The bowling object should know that when we've finished a throw, we move on to the next one. 
- the fact that we need to do all this setup in the first place feels a lot harder than if we just created a bowling object and injected a scorecard which was already initialised with a spare.

If we address all of these problems then our test looks like

```
//Arrange
var frame = new Frame(5, 5);
var scoredCard = new ScoredCard([frame]);
var bowling = new Bowling(scorecard);

// Act
bowling.throw(5);

// Assert
expect(... test condition...)
```